src/xo-app/xoa-updates/index.js

```diff
@@ -0,0 +1,279 @@
+import ActionButton from 'action-button'
+import ansiUp from 'ansi_up'
+import assign from 'lodash/assign'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import xoaUpdater, { exposeTrial, isTrialRunning } from 'xoa-updater'
+import { confirm } from 'modal'
+import { connectStore } from 'utils'
+import { FormattedDate } from 'react-intl'
+import { Password } from 'form'
+import isEmpty from 'lodash/isEmpty'
+import { error } from 'notification'
+
+xoaUpdater.start()
+
+const states = {
+  disconnected: 'Disconnected',
+  updating: 'Updating',
+  upgrading: 'Upgrading',
+  upToDate: 'Up to Date',
+  upgradeNeeded: 'Upgrade required',
+  registerNeeded: 'Registration required',
+  error: 'An error occured'
+}
+
+@connectStore((state) => {
+  return {
+    configuration: state.xoaConfiguration,
+    log: state.xoaUpdaterLog,
+    registration: state.xoaRegisterState,
+    state: state.xoaUpdaterState,
+    trial: state.xoaTrialState
+  }
+})
+export default class XoaUpdates extends Component {
+  constructor () {
+    super()
+    this.state = {}
+  }
+
+  // These 3 inputs are "controlled" http://facebook.github.io/react/docs/forms.html#controlled-components
+  _handleProxyHostChange = event => this.setState({proxyHost: event.target.value || ''})
+  _handleProxyPortChange = event => this.setState({proxyPort: event.target.value || ''})
+  _handleProxyUserChange = event => this.setState({proxyUser: event.target.value || ''})
+
+  _handleConfigReset = () => {
+    const { configuration } = this.props
+    const { proxyPassword } = this.refs
+    proxyPassword.value = ''
+    this.setState(configuration)
```

Comment:
Be careful: `setState()` does not set a new state, it merges (shallowly) the new state object with the previous one.

Also, you may want to namespace your state values.